EIPS/eip-747.md

```diff
@@ -31,68 +38,105 @@ In the second case, the user experience is terrible.
 By leveraging a user's existing trust with websites they are learning about assets on, we are able to decentralize the responsibility of managing a user's list of known assets.
 
 ## Specification
+
 <!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (go-ethereum, parity, cpp-ethereum, ethereumj, ethereumjs, and [others](https://github.com/ethereum/wiki/wiki/Clients)).-->
-A new method is added to web3 browsers' ethereum providers:
 
-```javascript
+We introduce the `wallet_watchAsset` RPC method:
 
-/**
-* @param  {Object}  opts - The options specifying the asset `type` and `options` specific for each of asset.
-* @returns  {Promise} success - Whether the user added the asset to their wallet.
-*/
-async  function  wallet_watchAsset (
-  opts
-) {
-/* Implementation would go here */
-}
+### wallet_watchAsset
 
-// Sample usage:
-web3.wallet.watchAsset({ type, options })
-```
-As there are several types of different assets, this method has to provide support for each of them in a separate way. If it doesn't, it should give a response according to that.
+Requests that a specified asset be added to the user's wallet, and returns `true` if the asset was successfully added, or an error if it was not.
```

Comment:
The current implementation in the MetaMask extension returns `false` if the user rejects the suggestion. Was changing the return in this case intentional? Which error would be returned if the user rejected the suggestion?

Reply:
Hm, we should refer to 1193 for the errors: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1193.md#provider-errors

I'll update that.